internal/mode/static/telemetry/collector.go

```diff
@@ -0,0 +1,78 @@
+package telemetry
+
+import (
+	"context"
+
+	v1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/nginxinc/nginx-gateway-fabric/internal/mode/static/state"
+)
+
+type Collector interface {
+	CollectData() Data
+}
+
+type DataCollector struct {
+	ctx       context.Context
+	k8sClient client.Client
+	processor state.ChangeProcessor
+}
+
+type GraphResourceCount struct {
+	Gateways           int
+	GatewayClasses     int
+	HTTPRoutes         int
+	ReferencedSecrets  int
+	ReferencedServices int
+}
+
+func NewDataCollector(
+	ctx context.Context,
+	k8sClient client.Client,
+	processor state.ChangeProcessor,
+) *DataCollector {
+	return &DataCollector{
+		ctx:       ctx,
+		k8sClient: k8sClient,
+		processor: processor,
+	}
+}
+
+func (c DataCollector) CollectData() Data {
```

Comment:
Any thoughts on having this method and having it call the other functions that each get one item in the data field?

Reply:
this seems reasonable.
for testing, I would test everything via CollectData() method. So that if we change/refactor "having it call the other functions that each get one item in the data field", we don't need to update the tests.